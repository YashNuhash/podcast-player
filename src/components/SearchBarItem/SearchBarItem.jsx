import React from 'react'
import styles from './SearchBarItem.module.scss'

const SearchBarItem = ({
    id,
    thumbnail,
    publisher_original,
    title_original,
}) => (
    <li className={styles.Item} key={id}>
        <a className={styles.Link} href=''>
            <img className={styles.Image} src={thumbnail} alt='' />
            <div>
                <h4 className={styles.Title}>{title_original}</h4>
                <p className={styles.Publisher}>{publisher_original}</p>
            </div>
        </a>
    </li>
)

export default SearchBarItem
