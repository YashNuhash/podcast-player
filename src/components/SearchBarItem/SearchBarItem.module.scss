@import '../../utils';

.Item {
    list-style: none;

    &:hover {
        background-color: $bg;
    }
}

.Link {
    padding: 1rem;
    display: flex;
    align-items: center;
    width: 100%;
    text-decoration: none;
    color: $text;
    transition: background-color 0.15s ease-in-out;

    &:focus,
    &:hover {
        outline: 0;
        background-color: $bg-2;
    }
}

.Image {
    width: 4rem;
    height: 4rem;
    margin-right: 1.5rem;
    flex-shrink: 0;
}

.Title,
.Publisher {
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
    text-overflow: ellipsis;
    white-space: normal;
    overflow: hidden;
}

.Title {
    color: $text;
    font-size: 1.5rem;
}

.Publisher {
    color: $light-text;
    font-size: 1.4rem;
}
