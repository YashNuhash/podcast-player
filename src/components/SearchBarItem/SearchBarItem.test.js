import React from 'react'
import { shallow } from 'enzyme'
import { SearchBarItem } from './SearchBarItem'

describe('SearchBarItem', () => {
    it('should render when there is no search term', () => {
        const mockPodcast = {
            id: '1',
            thumbnail: 'http://www.example.com/link',
            publisher_original: 'John Doe',
            title_original: 'Podcast',
            clearSearch: jest.fn(),
        }

        expect(shallow(<SearchBarItem {...mockPodcast} />)).toMatchSnapshot()
    })
})
