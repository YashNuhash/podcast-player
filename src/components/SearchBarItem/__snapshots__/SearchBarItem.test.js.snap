// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBarItem should render when there is no search term 1`] = `
<li
  className="Item"
  key="1"
>
  <Link
    className="Link"
    onClick={[Function]}
    to="/podcast/1"
  >
    <img
      alt=""
      className="Image"
      src="http://www.example.com/link"
    />
    <div>
      <EllipsisText
        className="Title"
        tagName="h4"
      >
        Podcast
      </EllipsisText>
      <EllipsisText
        className="Publisher"
        tagName="p"
      >
        John Doe
      </EllipsisText>
    </div>
  </Link>
</li>
`;
